packages/react/slot/src/Slot.tsx

```diff
@@ -11,63 +11,24 @@ interface SlotProps {
 
 const Slot = React.forwardRef<HTMLElement, SlotProps>((props, forwardedRef) => {
   const { children, ...slotProps } = props;
-  const childLength = React.Children.count(children);
-
-  if (childLength === 1) {
-    return (
-      <SlotClone {...slotProps} ref={forwardedRef}>
-        {children}
-      </SlotClone>
-    );
-  }
-
-  return (
-    <>
-      {React.Children.map(children, (child) => {
-        if (React.isValidElement(child) && child.type === Slottable) {
-          return (
-            <SlotClone {...slotProps} ref={forwardedRef}>
-              {child.props.children}
-            </SlotClone>
-          );
-        }
-
-        return child;
-      })}
-    </>
-  );
-});
-
-Slot.displayName = 'Slot';
-
-/* -------------------------------------------------------------------------------------------------
- * SlotClone
- * -----------------------------------------------------------------------------------------------*/
-
-interface SlotCloneProps {
-  children: React.ReactNode;
-}
-
-const SlotClone = React.forwardRef<any, SlotCloneProps>((props, forwardedRef) => {
-  const { children, ...slotProps } = props;
-  const child = React.Children.only(children);
-
-  return React.isValidElement(child)
-    ? React.cloneElement(child, {
-        ...mergeProps(slotProps, child.props),
-        ref: composeRefs(forwardedRef, (child as any).ref),
-      })
+  return children
+    ? React.Children.only(children) && React.isValidElement(children)
+      ? React.cloneElement(children, {
+          ...mergeProps(slotProps, children.props),
+          ref: composeRefs(forwardedRef, (children as any).ref),
+        })
+      : null
     : null;
 });
 
-SlotClone.displayName = 'SlotClone';
+Slot.displayName = 'Slot';
 
 /* -------------------------------------------------------------------------------------------------
  * Slottable
  * -----------------------------------------------------------------------------------------------*/
 
 const Slottable = ({ children }: { children: React.ReactNode }) => {
-  return children as React.ReactElement;
+  return <>{children}</>;
```

Comment:
I also noticed that components using `Slottable` (e.g. `Tooltip.Content`) would throw an error if there were no children, e.g. 

```jsx
<Tooltip.Content></Tooltip.Content>
```

Slipped this fix in while I was here. Hope that's okay 🙏